### PR TITLE
[T-6540][17.0][ADD] low_sold_product_report

### PR DESCRIPTION
**Pull request by Tisho99**

### CODE REVIEW

**Comment by ValentinVinagre** [comment]
@luis-ron could u review please?

**Review by luis-ron — APPROVED** [review]
Functional Review: LGTM 👍🏻

File: low_sell_products_report/report/low_product_report_views.xml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                        string="Product Type"
+                        name="Type"
+                        context="{'group_by':'detailed_type'}"
+                    />
+                </group>
+            </search>
+        </field>
+    </record>
+
+    <!-- <record id="action_order_report_all" model="ir.actions.act_window">
```
Comment (ValentinVinagre) [inline low_sell_products_report/report/low_product_report_views.xml]: Be careful when uploading unused code.

File: low_sell_products_report/report/low_sell_products_report.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
```
Comment (ValentinVinagre) [inline low_sell_products_report/report/low_sell_products_report.py]: delete

File: low_sell_products_report/report/low_sell_products_report.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
```
Comment (ValentinVinagre) [inline low_sell_products_report/report/low_sell_products_report.py]: copyright?

File: low_sell_products_report/security/low_sell_products_report_security.xml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+<!-- Copyright 2024 Alberto Martínez <alberto.martinez@sygel.es>
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+    <record id="low_sell_products_report_manager" model="res.groups">
+        <field name="name">Low Sell Products Report Access</field>
+        <field name="category_id" ref="base.module_category_hidden" />
+        <field
+            name="users"
+            eval="[(4, ref('base.user_root')), (4, ref('base.user_admin'))]"
```
Comment (ValentinVinagre) [inline low_sell_products_report/security/low_sell_products_report_security.xml]: thanks ❤️

File: low_sell_products_report/tests/__init__.py [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
```
Comment (ValentinVinagre) [inline low_sell_products_report/tests/__init__.py]: copyright

File: low_sell_products_report/wizards/low_sell_products_report_view.xml [hunk 6]
```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2024 Alberto Martínez <alberto.martinez@sygel.es>
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+<record id="low_sell_products_report_wizard_form" model="ir.ui.view">
```
Comment (ValentinVinagre) [inline low_sell_products_report/wizards/low_sell_products_report_view.xml]: Spaces need to be added

**Review by ValentinVinagre — CHANGES_REQUESTED** [review]
some comments

File: low_sell_products_report/README.rst [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+.. |badge2| image:: https://img.shields.io/badge/licence-AGPL--3-blue.png
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+.. |badge3| image:: https://img.shields.io/badge/github-sygel--technology%2Fsy--sale--reporting-lightgray.png?logo=github
+    :target: https://github.com/sygel-technology/sy-sale-reporting/tree/17.0/low_sell_products_report
+    :alt: sygel-technology/sy-sale-reporting
+
+|badge1| |badge2| |badge3|
+
+This module adds a report to query your less seld products, setteale
```
Comment (manuelregidor) [inline low_sell_products_report/README.rst]: "least sold products"
I think "setteale" is some kind of typo.

File: low_sell_products_report/README.rst [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+   the default values for the low sales report.
+
+Usage
+=====
+
+To use this module, you need to:
+
+1. Go to Sales / Reporting /Low sell products report
+2. Change the default values if needed, and click the Accept button
+3. You will see a list of the less sold products, given your date range
```
Comment (manuelregidor) [inline low_sell_products_report/README.rst]: "least sold products"

File: low_sell_products_report/__manifest__.py [hunk 9]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2024 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Low Sell Products Report",
+    "summary": "Adjustable Report of less sold product",
```
Comment (manuelregidor) [inline low_sell_products_report/__manifest__.py]: "least sold product"

File: low_sell_products_report/__manifest__.py [hunk 10]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2024 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Low Sell Products Report",
```
Comment (manuelregidor) [inline low_sell_products_report/__manifest__.py]: I don't think that "low sell" is correct.

File: low_sell_products_report/models/res_company.py [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+            ("1", "Last Month"),
+            ("3", "Last 3 Months"),
+            ("6", "Last 6 Months"),
+            ("12", "Last 12 Month"),
+        ],
+        string="Default analyzed period",
+        required=True,
+        default="1",
+    )
+    low_sales_report_default_sold_quantity_limit = fields.Float()
```
Comment (manuelregidor) [inline low_sell_products_report/models/res_company.py]: Shouldn't you limit this float to the number of digits for product quantity used in the system?

File: low_sell_products_report/models/res_company.py [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+    low_sales_report_default_sold_amount_limit = fields.Float()
```
Comment (manuelregidor) [inline low_sell_products_report/models/res_company.py]: Same here

File: low_sell_products_report/models/res_config_settings.py [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+            ("1", "Last Month"),
+            ("3", "Last 3 Months"),
+            ("6", "Last 6 Months"),
+            ("12", "Last 12 Month"),
+        ],
+        related="company_id.low_sales_report_default_period",
+        readonly=False,
+        required=True,
+    )
+    low_sales_report_default_sold_quantity_limit = fields.Float(
```
Comment (manuelregidor) [inline low_sell_products_report/models/res_config_settings.py]: Shouldn't you limit this float to the number of digits for product quantity used in the system?

File: low_sell_products_report/models/res_config_settings.py [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+        related="company_id.low_sales_report_default_sold_quantity_limit",
+        readonly=False,
+    )
+    low_sales_report_default_sold_amount_limit = fields.Float(
```
Comment (manuelregidor) [inline low_sell_products_report/models/res_config_settings.py]: Shouldn't you limit this float to the number of digits for product quantity used in the system?

File: low_sell_products_report/report/low_sell_products_report.py [hunk 15]
```diff
@@ -0,0 +1,162 @@
...
+            rec.name = (
+                rec.product_id.name
+                if rec.query_type == "product_product"
+                else rec.product_tmpl_id.name
+            )
+
+    def _select(self):
+        template_type = self.env.context.get("product_type") == "product_template"
+        select_ = f"""
+            MIN(l.id) AS id,
```
Comment (manuelregidor) [inline low_sell_products_report/report/low_sell_products_report.py]: Shouldnt the id be set automatically?

File: low_sell_products_report/wizards/low_sell_products_report.py [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+            datetime.date.today()
+            - relativedelta(
+                months=int(self.env.company.low_sales_report_default_period)
+            )
+        ),
+        required=True,
+    )
+    date_end = fields.Date(default=datetime.date.today(), required=True)
+
+    sold_quantity_limit = fields.Float(
```
Comment (manuelregidor) [inline low_sell_products_report/wizards/low_sell_products_report.py]: Shouldn't you limit this float to the number of digits for product quantity used in the system?

File: low_sell_products_report/wizards/low_sell_products_report.py [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+        default=lambda s: s.env.company.low_sales_report_default_sold_quantity_limit,
+    )
+    sold_amount_limit = fields.Float(
```
Comment (manuelregidor) [inline low_sell_products_report/wizards/low_sell_products_report.py]: Shouldn't you limit this float to the number of digits for product quantity used in the system?

**Review by manuelregidor — COMMENTED** [review]
Just a few comments. Also, we should try not to use plurals in module names (low_sell_product_report instead of low_sell_products_report).

File: low_sell_products_report/report/low_sell_products_report.py (hunk 15 above)
Comment (Tisho99) [reply low_sell_products_report/report/low_sell_products_report.py]: No, not in this type of model

File: low_sell_products_report/wizards/low_sell_products_report.py (hunk 16 above)
Comment (Tisho99) [reply low_sell_products_report/wizards/low_sell_products_report.py]: I dont see where the number of digits for the product quantity is declared

https://github.com/odoo/odoo/blob/06f18f44507553bfdd292a8a36762e2d3ad93a30/addons/sale/models/sale_order_line.py#L116

File: low_sell_products_report/wizards/low_sell_products_report.py (hunk 17 above)
Comment (Tisho99) [reply low_sell_products_report/wizards/low_sell_products_report.py]: I dont see where the number of digits for the product quantity is declared

https://github.com/odoo/odoo/blob/06f18f44507553bfdd292a8a36762e2d3ad93a30/addons/sale/models/sale_order_line.py#L154

**Review by manuelregidor — APPROVED** [review]
LGTM

**Review by ValentinVinagre — APPROVED** [review]

**Pull request merged**